rustls-platform-verifier/src/android.rs

```diff
@@ -165,17 +166,16 @@ impl<'a> Context<'a> {
 /// are cleared.
 pub(super) fn with_context<F, T>(f: F) -> Result<T, Error>
 where
-    F: FnOnce(&Context) -> Result<T, Error>,
+    F: FnOnce(&mut Context) -> Result<T, Error>,
 {
-    let context = global().context()?;
-    let env = context.env();
+    let mut context = global().context()?;
 
     // 16 is the default capacity in the JVM, we can make this configurable if necessary
-    env.push_local_frame(16)?;
+    context.env.push_local_frame(16)?;
 
-    let res = f(&context);
+    let res = f(&mut context);
 
-    env.pop_local_frame(JObject::null())?;
+    unsafe { context.env.pop_local_frame(&JObject::null()) }?;
```

Comment:
This should have a `// SAFETY:` comment attached to it now, but I'm attempting to determine the soundness requirements. [The documentation](https://docs.rs/jni/latest/jni/struct.JNIEnv.html#method.pop_local_frame) says local references become invalid, and I assume those are JVM references and not anything related to native code.

Given that we only return native Rust objects from `f`, I think we follow that rule correctly. If you happen to know, does that sound reasonable to you as well?

Reply:
Yes, I agree. Maybe it's better to declare `T: 'static` to make sure it's native rust object?